auth.php

```diff
@@ -545,6 +553,51 @@ public function saml_login() {
         return;
     }
 
+    /**
+     * Checks if the flagged user feature is enabled and if so, handles configured flag attribute
+     *
+     * @param $attributes
+     *
+     * @throws \moodle_exception
+     */
+    protected function handle_flagged_login($attributes) {
```

Comment:
Don't you think this function needs a bit of love? 
Even with existing comments it's really hard to follow what's going on there. 
Basically we need:
1. Check if  flagattribute set at all in configs. If not - > log, terminate handle. 
2. Check if flagattribute is exist in $attributes at all.  If not - > terminate handle.
3. Get flagattribute from $attributes and check if it's the same as configured value.
4, Then handle redirect based on isflagactive and configured flagresponsetype.

"1" and "2" could be abstracted to a validation function with a meaningful name.
"3" could be abstracted to  a function with a meaningful name.

This I recon will make the code much cleaner.

Reply:
Refactored into smaller methods with individual concerns, could you please review changes @dmitriim ?